### ISSUE TITLE
SSL/TLS for incoming connections

### ISSUE THREAD

**Issue by xmrig** [xmrig/xmrig-proxy#275]
SSL/TLS support for incoming connections added in [dev](https://github.com/xmrig/xmrig-proxy/commits/dev) branch.

* Documentation https://xmrig.com/docs/proxy/tls
* Please note, TLS increase CPU, memory and network load.

### Deprecation
Old style `bind` format:
```
    "bind": [
        "0.0.0.0:3333",
        "[::]:3333"
    ],
```
New style `bind` format:
```
    "bind": [
        {
            "host": "0.0.0.0",
            "port": 3333,
            "tls": false
        },
        {
            "host": "::",
            "port": 3333,
            "tls": false
        }
    ],
```
Old format still supported, but for new setups recommend use new format, also with old format not possible add TLS port.
Command line option `--bind` not deprecated, for TLS support added new option `--tls-bind` with same format.

**Comment by xmrig** [comment]
Option `protocols` changed to more human friendly, instead of JSON array `[ "TLSv1", "TLSv1.1", "TLSv1.2", "TLSv1.3" ]` just simple string `"TLSv1 TLSv1.1 TLSv1.2 TLSv1.3"`.

**Comment by lisergey** [comment]
@xmrig, is it possible to intermix TLS-enabled and plain ports in one process of xmrig-proxy, using different ports?

Like in config
```
    "bind": [
        {
            "host": "0.0.0.0",
            "port": 3333,
            "tls": false
        },
        {
            "host": "0.0.0.0",
            "port": 3334,
            "tls": true
        }
    ],
```
Thank you.

**Comment by xmrig** [comment]
@xmrig Yep, exactly that way.
Thank you.

**Comment by lisergey** [comment]
@xmrig, would do you think about making next release from dev branch?
it works nice and stable, so why not to make next step forward?

**Comment by xmrig** [comment]
Yep, you're right, I will make release in next few days.
Thank you.

**Comment by xmrig** [comment]
Included into v2.9.1 release.
Thank you.

**Comment by ttsite** [comment]
Xmrig whether to enable TLS at the same time ？
